README.md

```diff
@@ -1,2 +1,161 @@
+    - Click the green `Create App` button
+1. Click `Add features and functionality`, then `Permissions`
+1. Scroll down to `Scopes`, and add the following three permissions:
+    - `chat:write:bot`
+    - `channels:read`
+    - `users:read`
+
+    ![Add Scopes](images/slack-auth-setup-add-scopes.png)
+    - Click the green `Save Changes` button
+1. Install and authorize the app:
+    - Scroll to the top of the page and click the green `Install App to Workspace` button
+    - Check that your app's permissions are correct and click the green `Authorize` button
+
+    ![Authorize the App](images/slack-auth-setup-authorize.png)
+1. You should now see an `OAuth Access Token`. Copy this and paste it into your app's `.env` file.
+    ![OAuth Token](images/slack-auth-setup-token.png)
+    - The `.env` file should already be in the provided `.gitignore`, but it doesn't hurt to double check!
+
+Your application should now be able to access Slack!
+
+### Verification
+
+To verify that you've set up the slack token correctly, work with your partner to write a script that does the following:
+
+1. Use the `dotenv` gem to load environment variables
+1. Use HTTParty to send a GET request to the [`channels.list` endpoint](https://api.slack.com/methods/channels.list)
+1. Check that the request completed successfully, and print relevant information to the console if it didn't
+1. Loop through the results and print out the name of each channel
+
+If your tokens are set up correctly, each of you should be able to run this script.
+
+## Implementation Requirements
+
+We will do an in-class design activity for this project. While you should read through these requirements before that exercise, we do not recommend beginning work until we have finished classroom work.
+
+### Wave 1 - Listing Channels and Users
+
+As a user, when I launch this program... 
+
+- I should see information about how many channels and users were loaded
+- I should then be given three options for how to interact with the program:
+  - List users
+  - List channels
+  - Quit
+
+As a user who is at the program's input prompt...
+
+- When I type "list users", I should see a list of all the users in the Slack workspace. This list should include username, real name, and Slack ID.
+- When I type "list channels", I should see a list of all the channels for that workspace. This list should include the channel's name, topic, member count, and Slack ID.
+- When I type "quit", the program should exit.
+- After completing any command other than "quit", the program should reprint the list of commands and ask for another input.
+
+**Hint:** You may want to investigate the [Table Print gem](https://github.com/arches/table_print) to handle formatting tables.
+
+### Wave 2 - Selecting and Showing Details
+
+**Vocab:** channels and users are both _recipients_.
+
+As a user at the input prompt...
+
+- I should see three new options:
+  - Select user
+  - Select channel
+  - Details
```

Comment:
just to help clarify that there is one recipient that is either a user or a channel, could we make the options something like:
```
- Select a user as the recipient
- Select a channel as the recipient
- List recipient
```
or something like that?